pkg/vcs/git/commits.go

```diff
@@ -112,9 +112,18 @@ func HasCommitAfter(ctx context.Context, repo gitserver.Repo, date string, revsp
 	span.SetTag("RevSpec", revspec)
 	defer span.Finish()
 
+	if revspec == "" {
+		revspec = "HEAD"
+	}
+
+	commitid, err := ResolveRevision(ctx, repo, nil, revspec, nil)
```

Comment:
```suggestion
	// We resolve the revision first because some repositories unfortunately have *both* a HEAD branch
	// and remotes/origin/HEAD -> master reference. We've only encountered this on k8s.sgdev.org (see
	// https://sourcegraph.slack.com/archives/CHEKCRWKV/p1561004415100200?thread_ts=1560992541.075000&cid=CHEKCRWKV) and CommitCount below would fail to take into
	// account the first line that is output:
	//
	//  warning: refname 'HEAD' is ambiguous.
	// 
	// The reason we can't just solve this in CommitCount by handling the above output is because...
	commitid, err := ResolveRevision(ctx, repo, nil, revspec, nil)
```

Someone may look at this code in the future and think we can simplify it to just pass `revspec` into `CommitCount` below, so we should add a comment here.